Escape default arrays and sequences the same other default values

jacobperron opened the pull request:
Fixes #610

jacobperron commented:
As a regression test, I've added a new field to one of the test messages: https://github.com/ros2/test_interface_files/pull/16

jacobperron commented:
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=15332)](http://ci.ros2.org/job/ci_linux/15332/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=10043)](http://ci.ros2.org/job/ci_linux-aarch64/10043/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=13012)](http://ci.ros2.org/job/ci_osx/13012/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=15525)](http://ci.ros2.org/job/ci_windows/15525/)

jacobperron commented:
Rpr fails because https://github.com/ament/ament_cmake/pull/352 has not been released.

sloretz commented:
This is interesting. I can't predict if it will pass the `test_communication` tests. I recommend adding another test to test_interface_files with `"ハローワールド"` to see what it does when given a string with characters outside the latin-1 character set.

https://github.com/ros2/test_interface_files/blob/ea4d4f33eca97f37b4294e6ab012fa0f216de609/msg/WStrings.msg#L2-L4

IIUC `Hellö wörld!"` in that `.msg` file is encoded as the UTF-8 bytes `b'Hell\xc3\xb6 w\xc3\xb6rld!'`. When read as `latin-1` the string in Python would be very different from the original.

```
>>> b'Hell\xc3\xb6 w\xc3\xb6rld!'.decode('latin-1')
'HellÃ¶ wÃ¶rld!'
```

When [re-encoded as `latin-1`](https://github.com/ros2/rosidl/blob/36ed120f43daeaab31fd9ba2bf8dfb58db05091d/rosidl_adapter/rosidl_adapter/msg/__init__.py#L39) that should output the original bytes, but there's some more weirdness when the generators read the idl files.

https://github.com/ros2/rosidl/blob/36ed120f43daeaab31fd9ba2bf8dfb58db05091d/rosidl_parser/rosidl_parser/parser.py#L660

jacobperron commented:
> IIUC Hellö wörld!" in that .msg file is encoded as the UTF-8 bytes b'Hell\xc3\xb6 w\xc3\xb6rld!'. When read as latin-1 the string in Python would be very different from the original.

In fact there were test failures related to this that I missed locally: https://ci.ros2.org/job/ci_linux/15332/testReport/junit/rosidl_generator_py.test/test_interfaces/test_wstrings/

jacobperron committed:
Escape default arrays and sequences the same other default values

Fix #610

Apply the same encode/decode pattern and escaping as for other default values.

Signed-off-by: Jacob Perron <jacob@openrobotics.org>
